In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv(r"C:/Users/HP/OneDrive/Desktop/AI-Data-Science-Career-Intelligence/data/raw/ai_ds_job_salaries_2026.csv")

In [3]:
#Display the first few rows of the DataFrame
df.head()

,job_title,experience_level,employment_type,company_size,company_location,employee_residence,industry,remote_ratio,years_experience,education_level,...,ai_tools_hours_per_week,salary_currency,salary_usd,equity_offered_pct,bonus_pct,job_satisfaction_score,interviews_to_offer,switched_jobs_last_year,upskilling_hours_per_month,fears_ai_automation_score
0,Data Analyst,Lead,Part-time,S,IN,AU,Finance,0,14.4,Bachelors,...,12.5,INR,23844,0.505,11.6,5.5,8,True,2.2,7.4
1,Research Scientist,Senior,Full-time,L,US,US,Retail,50,7.6,Masters,...,2.1,USD,170132,0.054,2.9,7.0,3,False,8.2,5.3
2,Machine Learning Engineer,Entry,Full-time,M,US,US,Government,0,0.0,Bootcamp,...,5.1,USD,83059,0.191,7.5,5.9,1,False,2.8,6.8
3,Data Engineer,Senior,Full-time,M,BR,IN,Government,100,9.7,Masters,...,2.0,BRL,29531,0.102,9.3,5.6,5,False,7.8,5.0
4,Data Engineer,Mid,Full-time,M,US,US,Education,0,3.2,Bachelors,...,4.0,USD,70105,0.064,10.6,4.6,4,False,9.1,5.1


In [4]:
#Show the shape of the DataFrame
df.shape

(5000, 27)

In [5]:
#Looking at the column names
df.columns.tolist()

['job_title',
 'experience_level',
 'employment_type',
 'company_size',
 'company_location',
 'employee_residence',
 'industry',
 'remote_ratio',
 'years_experience',
 'education_level',
 'primary_language',
 'has_ml_in_title',
 'manages_people',
 'team_size',
 'certifications_count',
 'weekly_hours',
 'uses_ai_tools_daily',
 'ai_tools_hours_per_week',
 'salary_currency',
 'salary_usd',
 'equity_offered_pct',
 'bonus_pct',
 'job_satisfaction_score',
 'interviews_to_offer',
 'switched_jobs_last_year',
 'upskilling_hours_per_month',
 'fears_ai_automation_score']

In [6]:
#Display information about the DataFrame
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 27 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   job_title                   5000 non-null   object 
 1   experience_level            5000 non-null   object 
 2   employment_type             5000 non-null   object 
 3   company_size                5000 non-null   object 
 4   company_location            5000 non-null   object 
 5   employee_residence          5000 non-null   object 
 6   industry                    5000 non-null   object 
 7   remote_ratio                5000 non-null   int64  
 8   years_experience            5000 non-null   float64
 9   education_level             5000 non-null   object 
 10  primary_language            5000 non-null   object 
 11  has_ml_in_title             5000 non-null   bool   
 12  manages_people              5000 non-null   bool   
 13  team_size                   5000 

In [7]:
df.describe()

,remote_ratio,years_experience,team_size,certifications_count,weekly_hours,ai_tools_hours_per_week,salary_usd,equity_offered_pct,bonus_pct,job_satisfaction_score,interviews_to_offer,upskilling_hours_per_month,fears_ai_automation_score
count,5000.00000,5000.000000,5000.00000,5000.000000,5000.000000,5000.000000,5000.000000,5000.000000,5000.000000,5000.000000,5000.000000,5000.000000,5000.000000
mean,43.08000,8.361440,7.93620,1.820400,44.345400,8.437400,98605.406200,0.139020,10.802720,6.685320,5.230800,8.680740,4.846080
std,41.45428,4.820808,2.64793,1.344212,6.142761,5.503637,55369.310473,0.184243,5.152959,1.364302,2.412695,4.422549,1.725924
min,0.00000,0.000000,1.00000,0.000000,24.400000,0.000000,12000.000000,0.000000,0.000000,1.100000,1.000000,0.000000,1.000000
25%,0.00000,4.800000,6.00000,1.000000,40.100000,3.300000,55592.750000,0.017000,7.100000,5.800000,3.000000,5.600000,3.700000
50%,50.00000,8.100000,8.00000,2.000000,44.400000,8.500000,90572.000000,0.072000,10.600000,6.700000,5.000000,8.700000,4.800000
75%,100.00000,11.500000,10.00000,3.000000,48.500000,12.625000,129657.750000,0.175000,14.200000,7.600000,7.000000,11.700000,6.000000
max,100.00000,23.700000,19.00000,8.000000,67.300000,25.200000,372347.000000,1.264000,30.800000,10.000000,16.000000,26.200000,10.000000


In [8]:
df['experience_level'].value_counts()

experience_level
Mid          1613
Senior       1520
Entry         985
Lead          656
Executive     226
Name: count, dtype: int64

What this tells us

The dataset is mostly made up of Mid-level and Senior professionals:

Mid → 32.3%,
Senior → 30.4%,
Entry → 19.7%,
Lead → 13.1%,
Executive → 4.5%,

So later we can investigate questions like:

Does salary increase as experience level increases?

and:

Which experience level has the highest AI-tool adoption?

and:

Which experience level has the highest fear of AI automation?

In [9]:
df['job_title'].value_counts()

job_title
Data Scientist                   709
Data Engineer                    682
Data Analyst                     627
Machine Learning Engineer        593
AI Engineer                      506
Analytics Engineer               317
MLOps Engineer                   306
Business Intelligence Analyst    293
Research Scientist               269
LLM Engineer                     242
Data Science Manager             238
Computer Vision Engineer         218
Name: count, dtype: int64

In [10]:
df['company_location'].value_counts()

company_location
US    1786
IN     782
GB     488
DE     350
CA     350
AU     240
BR     210
NL     200
FR     189
PL     147
SG     130
ES     128
Name: count, dtype: int64

Later, we can investigate:
Which country has the highest average salary?
Does company location affect salary?
Which countries have the highest AI adoption?
Do employees living in a different country from their company tend to work remotely?

In [11]:
df['employment_type'].value_counts()

employment_type
Full-time    4159
Contract      418
Freelance     212
Part-time     211
Name: count, dtype: int64

So our dataset is heavily dominated by full-time workers.
Why this matters.Later, we can investigate questions such as:

Do full-time professionals earn more than contractors or freelancers?

and:

Does employment type affect job satisfaction or job switching?

and:

Which employment type has the highest AI-tool usage?


In [12]:
df['company_size'].value_counts()

company_size
M    2551
L    1288
S    1161
Name: count, dtype: int64

Why this is useful?

Later we can ask:

Do people at large companies earn more than people at small companies?

And:

Does company size affect AI adoption or job satisfaction?

In [13]:
df['employee_residence'].value_counts()

employee_residence
US    1796
IN     776
GB     489
DE     362
CA     354
AU     243
NL     206
BR     205
FR     173
PL     148
ES     129
SG     119
Name: count, dtype: int64

In [14]:
df['industry'].value_counts()

industry
Technology       1528
Finance           839
Healthcare        543
Retail            452
Consulting        442
Manufacturing     302
Media             268
Government        254
Education         196
Energy            176
Name: count, dtype: int64

What stands out?

Technology dominates the dataset, with about 31% of all records.

Finance is second at about 17%, followed by Healthcare at about 11%.

This is actually useful for our project because later we can investigate:

💰 Which industry pays the highest salaries?

🤖 Which industry uses AI tools the most?

😰 Which industry has the highest fear of AI automation?

🔄 Which industry has the highest job-switching rate?

📚 Which industry has the highest upskilling activity?

Those are exactly the kinds of questions that can turn this from a simple dataset analysis into a career intelligence project.

In [15]:
df['remote_ratio'].value_counts()

remote_ratio
0      2112
50     1468
100    1420
Name: count, dtype: int64

So:

42.2% are fully on-site,
29.4% are hybrid,
28.4% are fully remote

🔎 What this tells us! 
There is a fairly large mix of working arrangements. That's excellent for our project because we can later investigate:

💰 Does remote work affect salary?

😊 Are remote workers more satisfied?

🤖 Do remote workers use AI tools more frequently?

🔄 Are remote workers more likely to switch jobs?

These are good career intelligence questions.

In [16]:
df['years_experience'].describe()

count    5000.000000
mean        8.361440
std         4.820808
min         0.000000
25%         4.800000
50%         8.100000
75%        11.500000
max        23.700000
Name: years_experience, dtype: float64

The most important number here is the median
50% = 8.1 years

That means if we arrange all 5,000 people from least to most experienced, the person in the middle has about 8.1 years of experience.

Notice something interesting:

Mean   = 8.36
Median = 8.10

They're fairly close, suggesting that the experience distribution isn't extremely skewed.

🚨 One interesting thing

We have:

Minimum = 0
Maximum = 23.7

That doesn't necessarily mean there's a problem.

Someone with 0 years could be an entry-level professional.

And 23.7 years is plausible for a senior/lead/executive professional.

We don't clean or remove anything yet.

We'll investigate unusual values during our data-cleaning stage.

In [17]:
df['education_level'].value_counts()

education_level
Bachelors      1903
Masters        1786
PhD             535
Bootcamp        473
Self-taught     303
Name: count, dtype: int64

What stands out?

The majority have a university degree:

Bachelor's: 38.1%
Master's: 35.7%

Together:

73.8% of the dataset has either a Bachelor's or Master's degree.

But we also have:

Bootcamp graduates
Self-taught professionals
PhD holders

That's interesting for our career-intelligence project because we can eventually investigate:

💰 Does education level affect salary?

📈 Do self-taught professionals have similar salaries to degree holders?

🤖 Does education level affect AI-tool adoption?

🔄 Which education group switches jobs more frequently?

Again, don't analyze those yet. We're collecting our understanding first.

In [18]:
df['primary_language'].value_counts()

primary_language
Python    3129
SQL        923
R          290
Java       259
Scala      219
Julia      102
Rust        78
Name: count, dtype: int64

What stands out?

Python dominates the dataset:

About 63% of the professionals have Python as their primary language.

SQL is a distant second at about 18.5%.

That gives us an interesting career question for later:

Do Python-focused professionals earn more than professionals whose primary language is SQL, R, Java, etc.?

We can also investigate:

Which language is most common in each job title?
Which language is associated with the highest salary?
Does primary language relate to AI-tool usage?
Does language relate to years of experience?
Which languages are common among people who switched jobs?

These will be excellent EDA questions later.

In [19]:
df["has_ml_in_title"].value_counts()

has_ml_in_title
False    3172
True     1828
Name: count, dtype: int64

So about 36.6% of the records have an ML-related designation in the job title, while 63.4% don't.

Why this is interesting

Later, we can investigate:

Do people with ML in their job title earn more?

Do they have more years of experience?

Do they use AI tools more frequently?

Do they have a different fear of AI automation?

This could become one of our useful career intelligence features.

In [20]:
df["manages_people"].value_counts()

manages_people
False    3860
True     1140
Name: count, dtype: int64

In [21]:
df["team_size"].describe()

count    5000.00000
mean        7.93620
std         2.64793
min         1.00000
25%         6.00000
50%         8.00000
75%        10.00000
max        19.00000
Name: team_size, dtype: float64

In [22]:
df["certifications_count"].value_counts().sort_index()

certifications_count
0     797
1    1491
2    1352
3     791
4     379
5     132
6      49
7       8
8       1
Name: count, dtype: int64

In [23]:
df["certifications_count"].describe()

count    5000.000000
mean        1.820400
std         1.344212
min         0.000000
25%         1.000000
50%         2.000000
75%         3.000000
max         8.000000
Name: certifications_count, dtype: float64

In [24]:
df["certifications_count"].value_counts().sort_index()

certifications_count
0     797
1    1491
2    1352
3     791
4     379
5     132
6      49
7       8
8       1
Name: count, dtype: int64

What we can see

Most professionals have between 1 and 3 certifications.

The largest group is:

1 certification → 1,491 professionals

And only one professional has 8 certifications.

This is useful because later, when we do our cleaning and EDA, we may want to pay attention to the small groups at the upper end.

But we are not removing or changing anything yet.

In [25]:
df["weekly_hours"].describe()

count    5000.000000
mean       44.345400
std         6.142761
min        24.400000
25%        40.100000
50%        44.400000
75%        48.500000
max        67.300000
Name: weekly_hours, dtype: float64

What stands out?

The mean and median are almost identical:

Mean   = 44.35
Median = 44.40

So the distribution is fairly balanced around 44 hours per week.

The range is:

24.4 → 67.3 hours

Nothing immediately looks impossible, although the higher values may be worth examining later during data cleaning.

In [26]:
df["ai_tools_hours_per_week"].describe()

count    5000.000000
mean        8.437400
std         5.503637
min         0.000000
25%         3.300000
50%         8.500000
75%        12.625000
max        25.200000
Name: ai_tools_hours_per_week, dtype: float64

What stands out?

The typical professional spends around 8–8.5 hours per week using AI tools.

The range is:

0 → 25.2 hours/week

The median is 8.5 hours, slightly above the mean of 8.44, so there's no obvious major skew from these summary numbers alone.

One useful observation: 0 hours is valid here because some professionals don't use AI tools, so we shouldn't automatically treat zero as a missing value.

In [28]:
df[['salary_usd', 'equity_offered_pct', 'bonus_pct', 'job_satisfaction_score', 'interviews_to_offer', 'upskilling_hours_per_month', 'fears_ai_automation_score']].describe().T

,count,mean,std,min,25%,50%,75%,max
salary_usd,5000.0,98605.40620,55369.310473,12000.0,55592.750,90572.000,129657.750,372347.000
equity_offered_pct,5000.0,0.13902,0.184243,0.0,0.017,0.072,0.175,1.264
bonus_pct,5000.0,10.80272,5.152959,0.0,7.100,10.600,14.200,30.800
job_satisfaction_score,5000.0,6.68532,1.364302,1.1,5.800,6.700,7.600,10.000
interviews_to_offer,5000.0,5.23080,2.412695,1.0,3.000,5.000,7.000,16.000
upskilling_hours_per_month,5000.0,8.68074,4.422549,0.0,5.600,8.700,11.700,26.200
fears_ai_automation_score,5000.0,4.84608,1.725924,1.0,3.700,4.800,6.000,10.000


One thing we should investigate later

The equity_offered_pct column is interesting:

mean   = 0.139
median = 0.072
max    = 1.264

Its values are much smaller than the bonus percentages, so we'll need to understand exactly how that variable is represented before making a dashboard label such as "Equity %". We won't change it now.

The salary range is also quite wide:

$12,000 → $372,347

That makes salary an excellent candidate for our first visualizations and later outlier analysis.